### PR TITLE
add more coverage to travis

### PR DESCRIPTION
**Pull request by stephengroat**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+  - linux
+  - osx
+
+compiler:
+  - gcc
+  - clang
+
+before_install:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install xz; fi
```
Comment (Chocobo1) [inline .travis.yml]: why not merge into 1 single line

**Comment by stephengroat** [comment]
i tend to separate out into 2 because i've seen (very sporadic) issues due
to the `brew update`. whenever i have an issue with that step, i know that
i can just restart travis build and it usually works

**Comment by stephengroat** [comment]
@Chocobo1 this should be ready to go now.

`trusty` is now the travis default, it's no longer needed to explicitly list.

this pr will add `osx` and `gcc`/`clang` to the build matrix, testing them all

**Comment by rillian** [comment]
Merged as be336819acdc117711e879c7b4c00e877050df50, although I removed the homebrew invocation in a subsequent commit to speed up test feedback.

Thanks for the contribution!

**Pull request closed without merging**